src/libmongoc/tests/test-mongoc-sample-commands.c

```diff
@@ -3572,6 +3572,134 @@ callback (mongoc_client_session_t *session,
+    * members in the URI string; e.g.
+    * uri_repl = "mongodb://mongodb0.example.com:27017,mongodb1.example.com:" \
+    *    "27017/?replicaSet=myRepl";
+    * client = test_framework_client_new (uri_repl);
+    * For a sharded cluster, connect to the mongos instances; e.g.
+    * uri_sharded =
+    * "mongodb://mongos0.example.com:27017,mongos1.example.com:27017/";
+    * client = test_framework_client_new (uri_sharded);
+    */
+
+   client = get_client ();
+
+   mongoc_server_api_version_from_string ("1", &server_api_version);
+   server_api = mongoc_server_api_new (server_api_version);
+   mongoc_server_api_strict (server_api, true);
+
+   mongoc_client_set_server_api (client, server_api, &error);
+   /* End Versioned API Example 2 */
+}
+
+static void
+_test_sample_versioned_api_example_3 (void)
+{
+   /* Start Versioned API Example 3 */
+   mongoc_client_t *client = NULL;
+   mongoc_server_api_t *server_api = NULL;
+   mongoc_server_api_version_t server_api_version;
+   bson_error_t error;
+
+   /* For a replica set, include the replica set name and a seedlist of the
+    * members in the URI string; e.g.
+    * uri_repl = "mongodb://mongodb0.example.com:27017,mongodb1.example.com:" \
+    *    "27017/?replicaSet=myRepl";
+    * client = test_framework_client_new (uri_repl);
+    * For a sharded cluster, connect to the mongos instances; e.g.
+    * uri_sharded =
+    * "mongodb://mongos0.example.com:27017,mongos1.example.com:27017/";
+    * client = test_framework_client_new (uri_sharded);
+    */
+
+   client = get_client ();
+
+   mongoc_server_api_version_from_string ("1", &server_api_version);
+   server_api = mongoc_server_api_new (server_api_version);
+   mongoc_server_api_strict (server_api, false);
+
+   mongoc_client_set_server_api (client, server_api, &error);
+   /* End Versioned API Example 3 */
+}
+
+static void
+_test_sample_versioned_api_example_4 (void)
+{
+   /* Start Versioned API Example 4 */
+   mongoc_client_t *client = NULL;
+   mongoc_server_api_t *server_api = NULL;
+   mongoc_server_api_version_t server_api_version;
+   bson_error_t error;
+
+   /* For a replica set, include the replica set name and a seedlist of the
+    * members in the URI string; e.g.
+    * uri_repl = "mongodb://mongodb0.example.com:27017,mongodb1.example.com:" \
+    *    "27017/?replicaSet=myRepl";
+    * client = test_framework_client_new (uri_repl);
+    * For a sharded cluster, connect to the mongos instances; e.g.
+    * uri_sharded =
+    * "mongodb://mongos0.example.com:27017,mongos1.example.com:27017/";
+    * client = test_framework_client_new (uri_sharded);
+    */
+
+   client = get_client ();
+
+   mongoc_server_api_version_from_string ("1", &server_api_version);
+   server_api = mongoc_server_api_new (server_api_version);
+   mongoc_server_api_deprecation_errors (server_api, true);
+
+   mongoc_client_set_server_api (client, server_api, &error);
+   /* End Versioned API Example 4 */
```

Comment:
Needs a `mongoc_client_destroy (client)` and `mongoc_server_api_destroy (server_api)`.